app/Http/Controllers/Api/ApiController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+
+class ApiController extends Controller {
```

Comment:
Inheriting from this `ApiController` class will allow us to delete a LOT of code, because we've a lot of repeated code for doing JSON and the link headers.